frontend/src/components/ShootWorkers/ContainerRuntime.vue

```diff
@@ -69,22 +62,34 @@ export default {
     kubernetesVersion: {
       type: String,
       required: true
-    },
-    criNameIsRequired: {
-      type: Boolean,
-      default: false
     }
   },
   data () {
     return {
-      validationErrors,
-      valid: undefined
+      valid: undefined,
+      validationErrors
     }
   },
-  validations,
+  validations () {
+    return this.validators
+  },
   computed: {
-    criNames () {
-      return map(this.machineImageCri, 'name')
+    criItems () {
```

Comment:
I think it is wrong to do the validation in a getter and rely in the validator on that cched value for valid. Why do we need these items objects at all?
```html
    <v-select
      color="primary"
      item-color="primary"
      :items="criNames"
      :error-messages="getErrorMessages('criName')"
      @blur="$v.criName.$touch()"
      v-model="criName"
      label="Container Runtime"
    ></v-select>
```
```js
export default {
   ...
  data () {
    return {
      valid: undefined,
      validationErrors
    }
  },
  validations () {
    return this.validators
  },
  computed: {
    validCriNames () {
      return map(this.machineImageCri, 'name')
    },
    criNames () {
      const names = this.validCriNames
      if (!includes(names, this.criName)) {
        names.push(this.criName)
      }
      return names
    },
    criContainerRuntimeTypes () {
      const containerRuntime = find(this.machineImageCri, ['name', this.criName])
      const ociRuntimes = get(containerRuntime, 'containerRuntimes', [])
      return map(ociRuntimes, 'type')
    },
    criName: {
      get () {
        return get(this.worker, 'cri.name')
      },
      set (value) {
        this.$set(this.worker, 'cri', {
          ...this.worker.cri,
          name: value
        })
        this.selectedCriContainerRuntimeTypes = undefined
        this.$v.criName.$touch()
        this.validateInput()
      }
    },
    selectedCriContainerRuntimeTypes: {
      get () {
        const criContainerRuntimes = get(this.worker, 'cri.containerRuntimes')
        return criContainerRuntimes
          ? map(criContainerRuntimes, 'type')
          : undefined
      },
      set (value) {
        if (!isEmpty(value)) {
          const criContainerRuntimes = map(value, type => ({ type }))
          set(this.worker, 'cri.containerRuntimes', criContainerRuntimes)
        } else {
          unset(this.worker, 'cri.containerRuntimes')
        }
      }
    },
    validators () {
      return {
        criName: {
          validCriName: value => {
            return includes(this.validCriNames, value)
          }
        }
      }
    }
  },
  methods: {
    getErrorMessages (field) {
      return getValidationErrors(this, field)
    },
    validateInput () {
      if (this.valid !== !this.$v.$invalid) {
        this.valid = !this.$v.$invalid
        this.$emit('valid', { id: this.worker.id, valid: this.valid })
      }
    }
  },
  mounted () {
    this.$v.$touch()
    this.validateInput()
  },
  watch: {
    criNames () {
      this.$v.$touch()
      this.validateInput()
    }
  }
}
```

Reply:
changed the validation as suggested